src/main/java/school/hei/haapi/service/UserService.java

```diff
@@ -41,14 +41,15 @@ public class UserService {
   private final GroupService groupService;
   private final FileService fileService;
 
-  public FileHash uploadUserProfilePicture(byte[] bytes, String userId) {
+  public void uploadUserProfilePicture(MultipartFile profilePictureAsMultipartFile, String userId) {
     User user = findById(userId);
-    File tempFile = fileService.createTempFile(bytes);
+    File savedProfilePicture = fileService.getFileFromMultipartFile(profilePictureAsMultipartFile);
     String bucketKey =
-        getFormattedBucketKey(user, user.getFirstName()) + fileService.getFileExtension(tempFile);
+        getFormattedBucketKey(user, profilePictureAsMultipartFile.getOriginalFilename())
```

Comment:
store with ID not any other updatable field.

Also prefix with the file type ( for example PROFILE_PICTURE_uuid.jpg)